lib/os/json.c

```diff
@@ -627,10 +627,16 @@ static int64_t obj_parse(struct json_obj *obj, const struct json_obj_descr *desc
 	int64_t decoded_fields = 0;
 	size_t i;
 	int ret;
+	int depth = 1;
 
 	while (!obj_next(obj, &kv)) {
-		if (kv.value.type == JSON_TOK_OBJECT_END) {
-			return decoded_fields;
+		if (kv.value.type == JSON_TOK_OBJECT_START) {
+			depth++;
+		} else if (kv.value.type == JSON_TOK_OBJECT_END) {
+			depth--;
+			if (depth == 0) {
+				return decoded_fields;
+			}
 		}
```

Comment:
mh doesn't this cause fields within the object of the unsupported field to be parsed with the description of the outer object?
In other words, what about:
```json
{
    "invalid": {"valid":6},
    "valid": 5
}
```

I didn't test this but looking at the code it seems like the resulting value of "valid" would be `6` instead of the expected `5`.
You might be able to get around this by `continuing` the loop when `depth > 1`.

Reply:
You are right. I added a test that showcases this behavior. Very good eye as always.

Reply:
Yeah, fun fact: This problem isn't new and already existed before my other fixes. I added another commit that fixes that problem, too.